src/main/scala/TestConfigs.scala

```diff
@@ -53,39 +59,53 @@ class WithComparator extends Config(
       TileLinkParameters(
         coherencePolicy = new MESICoherence(site(L2DirectoryRepresentation)),
         nManagers = site(NBanksPerMemoryChannel)*site(NMemoryChannels) + 1,
-        nCachingClients = 1,
-        nCachelessClients = 1 + site(ComparatorKey).targets.size + site(ExtraL1Clients),
+        nCachingClients = site(NCachedTileLinkPorts),
+        nCachelessClients = site(NUncachedTileLinkPorts),
         maxClientXacts = 2,
         maxClientsPerPort = 1,
         maxManagerXacts = site(NAcquireTransactors) + 2,
         dataBeats = site(MIFDataBeats),
         dataBits = site(CacheBlockBytes)*8)
-    case BuildTiles =>
-      Seq((r: Bool, p: Parameters) => Module(new ComparatorTile(r)(p.alterPartial({case TLId => "L1toL2"}))))
+    case BuildTiles => {
+      val groundtest = if (site(XLen) == 64)
+        DefaultTestSuites.groundtest64
+      else
+        DefaultTestSuites.groundtest32
+      TestGeneration.addSuite(groundtest("p"))
+      TestGeneration.addSuite(DefaultTestSuites.emptyBmarks)
+      Seq((r: Bool, p: Parameters) => Module(new ComparatorTile(r)(
+        p.alterPartial({
+          case TLId => "L1toL2"
+          case NUncachedTileLinkPorts => 1 + site(ComparatorKey).targets.size
+        }))))
+    }
     case ComparatorKey => ComparatorParameters(
       targets    = Seq(0x80001000L, 0x80000000L),
       width      = 8,
       operations = 1000,
       atomics    = true)
-    case RoccNMemChannels => site(ComparatorKey).targets.size // Work-around bad rocket<>tile dependency
+    case NUncachedTileLinkPorts => 1 + site(ComparatorKey).targets.size
     case TohostAddr => BigInt("80001000", 16) // quit test by writing here
+    case UseFPU => false
+    case UseAtomics => false
     case _ => throw new CDEMatchError
   })
 
 class WithMemtest extends Config(
   (pname, site, here) => pname match {
-    case NGenerators => site(NTiles)
-    case GenerateUncached => true
-    case GenerateCached => true
+    case GroundTestCachedClients => 1
+    case GroundTestUncachedClients => 1
+    case GroundTestNPTW => 0
     case MaxGenerateRequests => 128
-    case GeneratorStartAddress => site(GlobalAddrMap)("mem").start
+    case GeneratorStartAddress => BigInt("80001040", 16)
```

Comment:
That sounds reasonable to me.  (Actually, TohostAddr could be rewritten as site(GlobalAddrMap)("mem").start + 0x1000 for increased clarity.)